src/CSE.cpp

```diff
@@ -33,6 +33,12 @@ bool should_extract(const Expr &e, bool lift_all) {
         return false;
     }
 
+    if (const Call *c = e.as<Call>()) {
```

Comment:
The need for this makes me think the extra args would be better just as flat extra args to the math intrinsic, instead of being packed into a struct.

Reply:
Well, if I don't pass the things as a `make_struct()`, CSE will still lift these arguments out of the call if you have multiple of those. This makes it only harder to figure out what the actual precision-arguments were to the Call node in the lowering pass that wants to read them back. They might now have become Let or LetStmt, instead of a simple ImmFloat.

Reply:
CSE will never lift constants, so they should be fine.